backend/simplemap/simplemap.go

```diff
@@ -0,0 +1,44 @@
+package simplemap
+
+import (
+	"github.com/heetch/confita"
+	"github.com/pkg/errors"
+	"reflect"
+
+	"context"
+)
+
+// Backend that loads config from map
+type Backend struct {
```

Comment:
How about just:

    type Backend map[string]interface{}

?
Then we wouldn't need the `NewBackend` constructor function.